keanu-project/src/main/java/io/improbable/keanu/algorithms/NetworkSamples.java

```diff
@@ -38,6 +39,26 @@ public NetworkSamples(Map<VertexId, ? extends List> samplesByVertex, List<Double
         this.size = size;
     }
 
+    public static NetworkSamples from(List<NetworkSample> networkSamples) {
+        Map<VertexId, List<?>> samplesByVertex = new HashMap<>();
```

Comment:
Now that you've created the `NetworkSample` class, can `samplesByVertex` become a `Map<VertexId, NetworkSample>`? That would be a lot cleaner.

Reply:
As a `NetworkSample` already maps from `VertexId` to a value for that vertex `Map<VertexId, NetworkSample>` wouldn't be needed. I think you may have meant a `List<NetworkSample>` instead. 

It might make it cleaner here. But then in methods such as `getDoubleTensorSamples` we would have to iterate through this list and extract the sample for a specific vertex. So we'd only be moving the complexity around.

Reply:
But it would mean we could delete the `logOfMasterPForEachSample` field

Reply:
After discussing - I think this is fine as it is.